src/component/Notations/PeakNotation.jsx

```diff
@@ -198,55 +192,23 @@ export const PeakNotation = ({
     }, 200);
   }, []);
 
-  const onMouseEnterAndLeaveHandler = useCallback(
-    (dispatchType) => {
-      if (spectrumID) {
-        dispatch({
-          type: dispatchType,
-          data: {
-            spectrumID: spectrumID,
-            objectType: 'peak',
-            objectID: id,
-          },
-        });
-      }
-    },
-    [dispatch, id, spectrumID],
-  );
-
-  const isHighlighted = useMemo(() => {
-    return isOver.flag === false &&
-      spectrumID &&
-      highlights &&
-      highlights[spectrumID].find(
-        (highlight) => highlight.type === 'peak' && highlight.id === id,
-      )
-      ? true
-      : false;
-  }, [highlights, id, isOver.flag, spectrumID]);
-
   return (
     <Fragment>
       <Global styles={styles} />
       <g
         id={xIndex}
         transform={`translate(${x}, ${y})`}
-        onMouseEnter={() => {
-          handleOnEnterNotation(xIndex);
-          onMouseEnterAndLeaveHandler(ADD_HIGHLIGHT);
-        }}
-        onMouseLeave={() => {
-          handleOnMouseLeaveNotation();
-          onMouseEnterAndLeaveHandler(DELETE_HIGHLIGHT);
-        }}
+        onMouseEnter={() => handleOnEnterNotation(xIndex)}
+        onMouseLeave={() => handleOnMouseLeaveNotation()}
+        {...highlight.onHover}
```

Comment:
this will override the callbacks set above. if you need custom handling, you have to do something like this:

```js
onMouseEnter={() => {
  handleOnEnterNotation(xIndex);
  highlight.show();
}
onMouseLeave={
  () => handleOnMouseLeaveNotation();
  highlight.hide();
}
```